base/operators.jl

```diff
@@ -525,8 +525,10 @@ julia> identity("Well, what did you expect?")
 """
 identity(x) = x
 
-+(x::Number) = x
-*(x::Number) = x
++(x) = x
+-(x) = (-1)*x
```

Comment:
Is `1` of the desired type here? Could `Int8(1)` reduce the risk of unwanted widening promotions? Or should the type depend on (the type of) `x` in some way? Then it is probably hard to write this generically.

Reply:
```suggestion
-(x) = Int8(-1)*x
```

Reply:
Yes, I agree. If we go with this approach, then the type should be as small as possible.